apps/src/blockly/addons/functionEditor.js

```diff
@@ -23,80 +111,237 @@ export default class FunctionEditor {
   // TODO
   refreshParamsEverywhere() {}
 
-  // TODO
-  openEditorForFunction(procedureBlock, functionName) {}
+  /**
+   * Show the given procedure in the function editor. Either load from
+   * the procedure workspace if it already exists, or create a new block.
+   * @param {Procedure} procedure The procedure to show.
+   */
+  showForFunction(procedure) {
+    Blockly.Events.disable();
```

Comment:
What is disabling and re-enabling before `clear` doing? Making sure that the deletion events aren't getting propagated to the other workspaces?

Reply:
yep, it's to prevent the events from propagating. I can add a comment making that clearer.